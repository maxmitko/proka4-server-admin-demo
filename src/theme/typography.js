
const defaultFontFamily = "'Roboto', sans-serif"

export default {
    h1: {
        fontWeight: 300,
        fontSize: 6,
        letterSpacing: -1.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h2: {
        fontWeight: 300,
        fontSize: 3.75,
        letterSpacing: -0.5,
        lineHeight: 1,
        fontFamily: defaultFontFamily,
        tag: "h2",
    },
    h3: {
        fontWeight: 400,
        fontSize: 3,
        letterSpacing: 0,
        lineHeight: 1.04,
        fontFamily: defaultFontFamily,
        tag: "h3",
    },
    h4: {
        fontWeight: 400,
        fontSize: 2.125,
        letterSpacing: 0.25,
        lineHeight: 1.17,
        fontFamily: defaultFontFamily,
        tag: "h4",
    },
    h5: {
        fontWeight: 400,
        fontSize: 1.5,
        letterSpacing: 0,
        lineHeight: 1.33,
        fontFamily: defaultFontFamily,
        tag: "h5",
    },
    h6: {
        fontWeight: 300,
        fontSize: 1.15,
        letterSpacing: 0,
        lineHeight: 1.4,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.15,
        lineHeight: 1.75,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    subtitle2: {
        fontWeight: 500,
        fontSize: 0.875,
        letterSpacing: 0.1,
        lineHeight: 1.57,
        fontFamily: defaultFontFamily,
        tag: "h6",
    },
    body1: {
        fontWeight: 400,
        fontSize: 1,
        letterSpacing: 0.5,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    body2: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 0.25,
        lineHeight: 1.5,
        fontFamily: defaultFontFamily,
        tag: "p",
    },
    button: {
        fontWeight: 400,
        fontSize: 0.875,
        letterSpacing: 1.25,
        lineHeight: 1.5,
        fontFamily: "'Roboto Condensed', sans-serif",
        tag: "span",
    },
    caption: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 0.4,
        lineHeight: 1.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    overline: {
        fontWeight: 400,
        fontSize: 0.75,
        letterSpacing: 1.5,
        lineHeight: 2.66,
        fontFamily: defaultFontFamily,
        tag: "span",
    },
    inherit: {
        fontWeight: "inherit",
        fontSize: "inherit",
        letterSpacing: "inherit",
        lineHeight: "inherit",
        fontFamily: "inherit",
        tag: "span",
    },
}