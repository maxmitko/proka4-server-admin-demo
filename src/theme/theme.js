import typography from './typography'
import produce from "immer"

export const theme = {
    palette: {
        primary: "hsla(0, 0%, 0%, 1)",
        secondary: "hsla(187, 100%, 42%, 1)",
        background: "hsla(0, 0%, 93%, 1)",
        surface: "hsla(0, 0%, 100%, 1)",
        error: "hsla(0, 100%, 42%, 1)",
    },
    paletteOn: {
        primary: "hsla(0, 0%, 100%, 1)",
        secondary: "hsla(0, 0%, 100%, 1)",
        background: "hsla(0, 0%, 0%, 1)",
        surface: "hsla(210, 9%, 31%, 1)",
        error: "hsla(0, 0%, 100%, 1)",
    },
    extra: {
        default: null,
        gray: "hsla(0, 0%, 76%, 1)",
        primary: "hsla(291, 64%, 42%, 1)",
        danger: "hsla(4, 90%, 58%, 1)",
        success: "hsla(123, 36%, 52%, 1)",
        warning: "hsla(34, 98%, 51%, 1)",
        info: "hsla(187, 100%, 42%, 1)",
        rose: "hsla(340, 82%, 58%, 1)",
        white: "hsla(0, 0%, 100%, 1)",
    },
    extraOn: {
        default: null,
        gray: "hsla(0, 0%, 100%, 1)",
        primary: "hsla(0, 0%, 100%, 1)",
        danger: "hsla(0, 0%, 100%, 1)",
        success: "hsla(0, 0%, 100%, 1)",
        warning: "hsla(0, 0%, 100%, 1)",
        info: "hsla(0, 0%, 100%, 1)",
        rose: "hsla(0, 0%, 100%, 1)",
        white: "hsla(0, 0%, 0%, 1)",
    },
    extraLight: {
        default: null,
        gray: "hsla(0, 0%, 72%, 1)",
        primary: "hsla(291, 88%, 57%, 1)",
        danger: "hsla(22, 100%, 53%, 1)",
        success: "hsla(123, 45%, 57%, 1)",
        warning: "hsla(34, 98%, 60%, 1)",
        info: "hsla(187, 95%, 46%, 1)",
        rose: "hsla(340, 82%, 65%, 1)",
        white: "hsl(0, 100%, 100%)",
    },
    extraShadow: {
        default: null,
        gray: "hsla(0, 0%, 30%, 0.22)",
        primary: "hsla(291, 64%, 54%, 0.25)",
        danger: "hsla(0, 100%, 58%, 0.25)",
        success: "hsla(123, 45%, 57%, 0.25)",
        warning: "hsla(34, 98%, 60%, 0.25)",
        info: "hsla(187, 95%, 46%, 0.25)",
        rose: "hsla(340, 82%, 63%, 0.25)",
        white: "hsla(0, 0%, 30%, 0.25)",
    },
    typography: typography,
    zIndex: {
        mobileStepper: 1000,
        appBar: 1100,
        drawer: 1200,
        modal: 1300,
        snackbar: 1400,
        tooltip: 1500,
    }
}

const newTheme = produce(theme, draftTheme => {
    draftTheme.extra.default = theme.extra.primary
    draftTheme.extraOn.default = theme.extraOn.primary
    draftTheme.extraLight.default = theme.extraLight.primary
    draftTheme.extraShadow.default = theme.extraShadow.primary
})


export default newTheme