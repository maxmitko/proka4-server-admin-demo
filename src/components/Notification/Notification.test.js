import { mount } from 'enzyme';
import React from 'react';
import { ThemeProvider } from 'styled-components'
import theme from 'theme/theme'
import TestComponent from './index'
import renderer from 'react-test-renderer';

const props = {
    text: 'test message',
    type: 'default',
    open: true,
    onClose: jest.fn(),
    autoClose: 0,
    hiddenButton: false,
    modal: false,
}

const withHOC = ui => (
    <ThemeProvider theme={theme}>
        {ui}
    </ThemeProvider >
)

describe('enzyme Paginator', () => {
    let wrapper;
    let element;

    beforeEach(() => {
        element = withHOC(<TestComponent {...props} />)
    })

    afterEach(() => {
        wrapper && wrapper.unmount();
    });

    it('default SNAPSHOT should render correctly', () => {
        const tree = renderer.create(element).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('check message should render correctly', () => {
        wrapper = mount(element);
        expect(wrapper.find('Text').html()).toMatch(/test message/);
    });

    it('check color render correctly', () => {
        wrapper = mount(element);
        const color = theme.extra.default.replace(/\s/g, '')
        expect(wrapper.find({ "data-test-id": "MessageStyled" })).toHaveStyleRule('background-color', color)
    });
});
