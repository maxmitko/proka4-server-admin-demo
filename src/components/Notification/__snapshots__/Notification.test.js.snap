// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme Paginator default SNAPSHOT should render correctly 1`] = `
.c1 {
  margin: 0;
  font-family: 'Roboto',sans-serif;
  font-size: 0.875rem;
  font-weight: 400;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  line-height: 1.5;
  text-align: inherit;
}

.c3 {
  border-radius: 0.2rem;
  cursor: pointer;
  min-height: auto;
  text-transform: uppercase;
  border: none;
  outline: none;
  color: hsla(0,0%,0%,1);
  font-family: 'Roboto Condensed',sans-serif;
  font-size: 0.875rem;
  -webkit-transition: 200ms cubic-bezier(0.4,0,0.2,1);
  transition: 200ms cubic-bezier(0.4,0,0.2,1);
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.875rem;
  padding: 12px 17px;
  background-color: inherit;
  color: hsla(0,0%,100%,1);
  border-radius: 50%;
  padding: 15px;
}

.c3:disabled,
.c3:hover:disabled {
  opacity: 0.75;
  cursor: auto;
  box-shadow: none;
}

.c3:hover {
  color: hsla(0,0%,100%,1);
  background-color: hsla(0,0%,100%,0.25);
}

.c3:disabled,
.c3:hover:disabled {
  background-color: inherit;
  color: hsla(0,0%,100%,1);
}

.c5 {
  position: absolute !important;
  border-radius: inherit;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c4 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  padding: 0 15px;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 0.2rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  min-width: 560px;
  min-height: 70px;
  background-color: hsla(291,64%,42%,1);
  color: hsla(0,0%,100%,1);
  box-shadow: 0 4px 20px 0px rgba(0,0,0,0.14), 0 10px 16px -5px hsla(291,64%,54%,0.25);
}

.c2 {
  padding: 4px;
}

.c2 svg {
  font-size: 18px;
}

<div
  className="c0"
  color="default"
  data-test-id="MessageStyled"
>
  <p
    className="c1"
    color="inherit"
  >
    test message
  </p>
  <button
    className="c2 c3"
    color="white"
    onClick={[MockFunction]}
    size="medium"
  >
    <div
      className="c4"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-1"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <div
      className="rtr-root c5"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
    >
      <span />
    </div>
  </button>
</div>
`;
